Announce.php

```diff
@@ -103,4 +105,82 @@ function schema() {
 			array( 'UpdateFunction', 'delete_orphans_dismissals' ),
 		);
 	}
+
+	/**
+	 * Add the RESTful routes handled by this plugin.
+	 *
+	 * @param string $p_event_name The event name
+	 * @param array  $p_event_args The event arguments
+	 * @return void
+	 */
+	public function routes( $p_event_name, $p_event_args ) {
+		$t_app = $p_event_args['app'];
+		$t_plugin = $this;
+		$t_app->group(
+			plugin_route_group(),
+			function() use ( $t_app, $t_plugin ) {
+				$t_app->get( '/dismiss/{context_id}', [$t_plugin, 'route_dismiss'] );
+			}
+		);
+	}
+
+	/**
+	 * RESTful route for Announcement dismissal.
+	 *
+	 * Returned JSON structure
+	 *   - {string} title
+	 *   - {string} text
+	 *
+	 * @param Psr\Http\Message\ServerRequestInterface $request
+	 * @param Psr\Http\Message\ResponseInterface $response
+	 * @param array $args
+	 * @return Psr\Http\Message\ResponseInterface
+	 */
+	public function route_dismiss( $request, $response, $args) {
+		$t_timestamp = time();
+		$t_user_id = auth_get_current_user_id();
+
+		# Set the reference Context Id for dismissal
+		$t_context_id = (int)$args['context_id'];
```

Comment:
Check that the context_id field was passed.

Reply:
Slim does that for me, AFAICT - the following request http://path.to/api/rest/plugins/Announce/dismiss (without providing a *context_id*) yields a 404

Reply:
Make sure to test this with `http://path.to/api/rest/plugins/Announce/dismiss/`.

Reply:
> Make sure to test this with http://path.to/api/rest/plugins/Announce/dismiss/

I did -
```console
$ curl -si -X POST http://localhost/mantis/api/rest/plugins/Announce/dismiss |head -1
HTTP/1.1 404 Not Found
$ curl -si -X POST http://localhost/mantis/api/rest/plugins/Announce/dismiss/ |head -1
HTTP/1.1 404 Not Found
```

> Also you may want to validate the input. For example, is the context_id > 0?

This is actually done by *AnnounceContext::load_by_id()*, 

```console
$ curl -si -X POST http://localhost/mantis/api/rest/plugins/Announce/dismiss/0 |head -1
HTTP/1.1 400 Bad Request
```